### PR TITLE
[US20] add Post entity, controller, service, DTO, repository, exception handling

### PR DESCRIPTION
**Pull request by Leonert**

### CODE REVIEW

File: social-network-api/src/main/java/com/socialnetwork/api/controller/PostController.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/posts")
```
Comment (CoffeeScriptHero) [inline social-network-api/src/main/java/com/socialnetwork/api/controller/PostController.java]:
```suggestion
@RequestMapping("/api/posts")
```

File: social-network-api/src/main/java/com/socialnetwork/api/controller/PostController.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    //Retrieving user id from DTO (only needed field in request), looking for it
+    // in userservice and throwing exception if user with such id wasn`t found
+    //Also checking of JWT token and comparing it with user needed to be realise in future TODO
+    Post post = convertToPost(postDto);
+    Optional<User> user = userService.findById(post.getUser().getId());
+    if (user.isEmpty()) {
+      throw new NoUserWithSuchCredentialsException();
+    }
+    post.setUser(user.get());
+    postService.save(convertToPost(postDto));
```
Comment (CoffeeScriptHero) [inline social-network-api/src/main/java/com/socialnetwork/api/controller/PostController.java]: postService.save(post);

Comment (Leonert) [reply social-network-api/src/main/java/com/socialnetwork/api/controller/PostController.java]: added

File: social-network-api/src/main/java/com/socialnetwork/api/controller/PostController.java (hunk 1 above)
Comment (Leonert) [reply social-network-api/src/main/java/com/socialnetwork/api/controller/PostController.java]: changed

**Review by CoffeeScriptHero — APPROVED** [review]

**Review by miegami — APPROVED** [review]

**Pull request merged**